### DISCUSSION TITLE
Idena Implementation Proposal: Mining Sessions

### DISCUSSION THREAD

**Discussion by simontakouridis** [idena-network/idena-go#666] [Ideas]
**Introduction:**

This proposal entails removing the requirement for all validated users to create flips, and instead implement incentive structures for users that enjoy making flips to produce good flips for all to consume. This will benefit Idena in 3 ways, firstly by having consistently high consensus flips for better network integrity, secondly by faster network growth because ordinary users are no longer discouraged by having mandatory flip creation steps, and thirdly it would be a disincentive for anyone wanting to create a flip making business that undermines Idena.

**Proposal in detail:**

Validation sessions:
- no longer require users to create flips.
- no longer provide the ability for users to mine.
- are limited in purpose to proving Idena users are human.

Mining sessions:
 (are similar to Validation sessions, with the following differences)
- Idena members are not required to participate
- sessions involve both flip solving and flip creating.
- sessions are frequent, e.g. daily.
- successful flip solvers are allowed to mine blocks till the next mining session.
- after several mining sessions, the top xx% of the best flip creators are tabulated. They exclusively have the opportunity to provide the flips for the next Validation session.

**Benefits in detail:**

1. Having consistently high consensus flips.

Currently Idena requires all members create flips. This poses an issue because flips are very difficult to create for an ordinary person. As such, flip quality will drop over time as the network grows to include more ordinary people. This results in lower consensus and places the integrity of Idena’s proof-of-person at risk.

Mining sessions redirects the process of flip creation to those users who enjoy it. These users are highly experienced in flip creation because they do it on a daily basis, and do it competitively. All successful flip creators are rewarded by having the ability to mine, however the best of the flip creators are further rewarded by having the opportunity of supplying the flips for the next Validation session. This results in Validation sessions being composed of extremely high quality flips with high consensus, thus strengthening Idena’s proof-of-person.

2. Faster network growth.

It is very difficult for ordinary users to create flips, and it therefore discourages users to continue their participation in Idena. Removing this requirement will help Idena retain most of its new users and therefore grow rapidly and sustainably.

3. Thwarts the “flip service” attack.

Requiring all users to create flips inadvertently creates a strong market demand for a flip service. Most people do not want to participate in this very cumbersome process and are therefore highly incentivised to outsource it. As such, by removing the flip creation process from user validation, the “flip service” provider market demand will decrease.

The “flip service” is also impeded by not being able to procure employees. This is because prospective employees are already directly rewarded by Idena via Mining sessions. There is no economic benefit for prospective employees to lower their own income and go through an intermediary “flip service” for it to generate a profit.

**Comment by cryptude** [comment]
I largely like the idea, but see two risks it introduces
1. If there is a consistent, small pool of top flip creators, the risk of collusion becomes much higher.
2. If a miner needs to interact with Idena on a daily basis, the number of people performing consensus would decrease, reducing Idena's USP as a highly decentralized blockchain. It may have the upside that farms become less profitable however, as you need constant labor to maintain mining rights.

**Reply by simontakouridis** [reply]
Great to hear. I also agree with your feedback. I think the best way to overcome these issues you outline is to simply wait for Idena to grow larger before implementing Mining Sessions. Perhaps 50-100K+ users?

**Reply by simontakouridis** [reply]
I've had some time now to reflect on these risks that you mentioned. So I will address solutions to these in more detail:

> 1. If there is a consistent, small pool of top flip creators, the risk of collusion becomes much higher.

These is an extremely serious risk, not because of collusion by a bunch of cheaters, but rather from a protest movement by miners that run counter to the demands of the general Idena community. Theoretically, a protest movement could bring the entire Idena network down. The following amendment to the original idea I put forth will counter this:

Original suggestion:
- after several mining sessions, the top xx% of the best flip creators are tabulated. They exclusively have the opportunity to provide the flips for the next Validation session.

Amended suggestion:
- after several mining sessions, the top xx% of the best flip creators are tabulated. They are better rewarded than the rest of the Idena community for providing flips for the next Validation session.

The key here is that all members of Idena can still provide flips for Validation sessions, if they wish to. As such, if miners organise a protest to disrupt the upcoming Validation session, the broader Idena community can step in to provide flips and alleviate the threat. This also goes with regular mining sessions, the broader Idena community can always step in and mine to relieve any slack.


> 2. If a miner needs to interact with Idena on a daily basis, the number of people performing consensus would decrease, reducing Idena's USP as a highly decentralised blockchain. It may have the upside that farms become less profitable however, as you need constant labor to maintain mining rights.

The risk of this is quite low when the Idena network has a vast membership across the globe. Even if just 1% of people participate in mining, in a network of 100K people, the network would still be significantly decentralised. However, when you consider how high unemployment is worldwide and only increasing due to AI, I would expect far more people to be participating. If Idena would like to increase the level of mining participants, the rewards could always be increased.

**Reply by cryptude** [reply]
Agree with the criticality of 1. - and the solution solves it; but it does also mean that the quality of the flips aren't guaranteed to be higher, as everyone will still try to get rewards. Maybe biasing the selection to the high quality flip creators might fix that, or allowing them to create more flips than the rest

2. agree that once the network is large, the problem dissipates, but getting to that scale means that you need a high percentage of the network interacting daily to maintain decentralization until that point.

**Reply by simontakouridis** [reply]
I really appreciate your critical analysis of the proposal.

> but it does also mean that the quality of the flips aren't guaranteed to be higher, as everyone will still try to get rewards. Maybe biasing the selection to the high quality flip creators might fix that, or allowing them to create more flips than the rest

The best approach is to simply reward the xx% best flip creators with a greater reward for the flip they submit for the Validation session. This results in a bias toward greater flip quality because:
1. Ordinary Idena members are not required to create flips, thus most will probably opt out of the process. (low probability of flip submission)
2. Some ordinary members may participate depending on the reward, and their confidence level in flip creation. (moderate probability of flip submission)
3. Miners probably will participate because they are experienced and capable flip creators. (high probability of flip submission)
4. The xx% best flip creators will almost certainly submit flips for the Validation process due to the higher reward provided. (Very high probability of flip submission)

As you can see, there is a loose correlation between: user capability to produce good quality flips, and the probability they will actually submit flips for the next Validation session. Therefore, the most capable flip creators submit, and the least capable do not submit flips for the next Validation sessions. Obviously the rewards can't be too high for the ordinary member submitting flips otherwise yes, everyone probably will submit.

NB: It would be a mistake to reward the xx% best flip creators with more flip submissions because a protest movement may involve purposeful submissions of faulty flips. Therefore it is imperative that the broader Idena membership is capable of thwarting this attack by submitting more flips to counteract when protests arise.

It's worth noting that if protests are erupting by the miners, they probably have valid reasons for being dissatisfied and these concerns should be addressed by the steering committee.
